daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -562,6 +562,14 @@ private[lf] final class Compiler(
         compile(env, body)
       case EFromRequiredInterface(requiredIfaceId @ _, requiringIfaceId, body @ _) =>
         SBFromRequiredInterface(requiringIfaceId)(compile(env, body))
+      case EInterfaceTemplateTypeRep(ifaceId, body @ _) =>
+        SBInterfaceTemplateTypeRep(ifaceId)(compile(env, body))
+      case ESignatoryInterface(ifaceId, body @ _) =>
+        val arg = compile(env, body)
+        SBSignatoryInterface(ifaceId)(arg, arg)
```

Comment:
I think this should do the trick so `SBResolveVirtual` gets the argument twice, but I'm not quite sure. Could you please tell me if this makes sense? @remyhaemmerle-da @sofiafaro-da @cocreature

Reply:
Looks sensible to me. We may want to see if passing the argument twice is really the best way to address this (for these an for other primitives) but definitely not in this PR.

Reply:
I think this is good enough for the MVP.